src/renderer.test.tsx

```diff
@@ -0,0 +1,447 @@
+        testHandlers(timeline);
+    })
+
+    test('ElementKind.ChartAtom', () => {
+        const nodes = render(chartElement())
+        const chart = nodes.flat()[0];
+        expect(getHtml(chart)).toContain('srcDoc=\"&lt;main&gt;Chart content&lt;/main&gt;\"');
+    })
+
+    
+
+    function testHandlers(node: ReactNode): void {
+        if (isValidElement(node)) {
+            const container = document.createElement("div");
+            document.body.appendChild(container);
+            act(() => {
+                renderDom(node, container);
+            });
+            const likeButton = container.querySelector('[data-testid="like"]');
+            const dislikeButton = container.querySelector('[data-testid="dislike"]');
+            const feedback = container.querySelector('[data-testid="feedback"]');
+            expect(feedback?.getAttribute('hidden')).toBe("")
+            dislikeButton.simulate('click');
+            likeButton.simulate('click');
+            expect(feedback.html()).not.toContain('hidden=""');
+            unmountComponentAtNode(container);
+            container.remove();
+        }
+    }
+});
+
+describe('Paragraph tags rendered correctly', () => {
+    test('Contains no styles in standfirsts', () => {
+        const fragment = JSDOM.fragment('<a href="#"><ul><li><strong><p>Standfirst link</p></strong></li></ul></a>');
+        const nodes = renderStandfirstText(fragment, mockFormat);
+        const html = getHtml(nodes.flat()[0]);
+        expect(html).toContain('<strong><p>Standfirst link</p></strong>');
+    });
+
+    test('Contains styles in article body', () => {
+        const fragment = JSDOM.fragment('<ul><li><strong><p>Standfirst link</p></strong></li></ul>');
+        const nodes = renderText(fragment, mockFormat);
+        const html = getHtml(nodes.flat()[0]);
+        expect(html).not.toContain('<strong><p>Standfirst link</p></strong>');
+    });
+});
+
+describe('Transforms hrefs', () => {
+    test('Transforms profile links', () => {
+       const href = "profile/firstname_lastname";
+       const transformed = transformHref(href);
+       expect(transformed).toBe("https://www.theguardian.com/profile/firstname_lastname")
+    });
+
+    test('Transforms latest links', () => {
+       const href = "https://www.theguardian.com/world/series/coronavirus-live/latest";
+       const transformed = transformHref(href);
+       expect(transformed).toBe("https://www.theguardian.com/world/series/coronavirus-live")
+    });
+
+    test('Does not transform valid link', () => {
+        const href = "https://www.theguardian.com/world/series/coronavirus-live";
+        const transformed = transformHref(href);
+        expect(transformed).toBe("https://www.theguardian.com/world/series/coronavirus-live")
+     });
+})
```

Comment:
It might be a good idea to write some tests to validate what you have done so far?

Reply:
@frankie297 a little stuck here on how to test using transformHref, any pointers?